### ISSUE TITLE
ProcessExecutor is not handling timeout correctly

### ISSUE THREAD

**Issue by yoosiba** [NumberFour/n4js#226]
`ProcessExecutor` is not handling timeout correctly, it will throw `IllegalThreadStateException` at https://github.com/NumberFour/n4js/blob/master/plugins/eu.numberfour.n4js.utils/src/eu/numberfour/n4js/utils/process/ProcessExecutor.java#L74

This causes exceptions :

```
06:23:02 Exception in thread "main" java.lang.IllegalThreadStateException: process hasn't exited
06:23:02    at java.lang.UNIXProcess.exitValue(UNIXProcess.java:424)
06:23:02    at eu.numberfour.n4js.utils.process.ProcessExecutor.execute(ProcessExecutor.java:83)
06:23:02    at eu.numberfour.n4js.utils.process.ProcessExecutor.createAndExecute(ProcessExecutor.java:153)
06:23:02    at eu.numberfour.n4js.binaries.BinaryCommandFactory$1.execute(BinaryCommandFactory.java:54)
06:23:02    at eu.numberfour.n4js.external.NpmManager.installPackage(NpmManager.java:455)
06:23:02    at eu.numberfour.n4js.external.NpmManager.installDependency(NpmManager.java:185)
06:23:02    at eu.numberfour.n4js.hlc.N4jsc.cloneGitRepositoryAndInstallNpmPackages(N4jsc.java:579)
06:23:02    at eu.numberfour.n4js.hlc.N4jsc.doMain(N4jsc.java:473)
06:23:02    at eu.numberfour.n4js.hlc.N4jsc.main(N4jsc.java:321)
```

Please handle timeout in more graceful way.

**Comment by lbeurerkellner** [comment]
This was more or less fixed by @mor-n4 with commit ffd5f8c9e0757cc1a0827c92f2916690f91d5df7. Due to an early return in case of a timeout, this exception can't be reproduced.

**Comment by lbeurerkellner** [comment]
After short discussion with @yoosiba, this issue was considered fixed.
